packages/commonwealth/client/scripts/views/modals/AuthModal/AuthModal.tsx

```diff
@@ -0,0 +1,160 @@
+};
+
+const AuthModal = ({
+  isOpen,
+  onClose,
+  onSuccess,
+  showWalletsFor,
+}: AuthModalProps) => {
+  const [activeTabIndex, setActiveTabIndex] = useState<number>(0);
+
+  const { wallets, onWalletSelect, onSocialLogin } = useWallets({
+    onModalClose: onClose,
+    onSuccess,
+  });
+
+  const tabsList = [
+    {
+      name: 'Wallet',
+      options: [
+        // Note: If `showWalletsFor` is present then show wallets for that chain, else
+        // only show ethereum based wallets options on non-community pages or on ethereum based communities
+        ...((!showWalletsFor
+          ? !app?.chain?.base
+          : showWalletsFor === ChainBase.Ethereum) ||
+        app?.chain?.base === ChainBase.Ethereum
+          ? ['walletConnect']
+          : []),
+        // only show cosmos based wallets on non-community pages or on cosmos based communities
+        ...(wallets || [])
+          .filter(
+            (wallet) =>
+              ((!showWalletsFor
+                ? !app?.chain?.base
+                : showWalletsFor === ChainBase.CosmosSDK) ||
+                app?.chain?.base === ChainBase.CosmosSDK) &&
+              wallet.chain === ChainBase.CosmosSDK,
+          )
+          .map((wallet) => wallet.name),
+        // only show solana based wallets on solana based communities
+        ...(wallets || [])
+          .filter(
+            (wallet) =>
+              app?.chain?.base === ChainBase.Solana &&
+              wallet.chain === ChainBase.Solana,
+          )
+          .map((wallet) => wallet.name),
+        // only show substrate based wallets on substrate based communities
+        ...(wallets || [])
+          .filter(
+            (wallet) =>
+              app?.chain?.base === ChainBase.Substrate &&
+              wallet.chain === ChainBase.Substrate,
+          )
+          .map((wallet) => wallet.name),
+      ],
+    },
+    {
+      name: 'Email or Social',
+      options: ['google', 'discord', 'x', 'github', 'email'],
+    },
+  ] as const;
+
+  const onAuthMethodSelect = async (option: AuthTypes) => {
+    if (option === 'email') {
+      // TODO: implement this in https://github.com/hicommonwealth/commonwealth/issues/6386
+      return;
+    }
+
+    // if any wallet option is selected
+    if (activeTabIndex === 0) {
+      await onWalletSelect(
+        wallets.find(
+          (wallet) => wallet.name.toLowerCase() === option.toLowerCase(),
+        ),
+      );
+    }
+
+    // if any SSO option is selected
+    if (activeTabIndex === 1) {
+      // TODO: decide if twitter references are to be updated to 'x'
+      await onSocialLogin(
+        option === 'x' ? WalletSsoSource.Twitter : (option as WalletSsoSource),
+      );
+    }
+  };
+
+  return (
+    <CWModal
+      open={isOpen}
+      onClose={onClose}
+      content={
+        <section className="AuthModal">
+          <CWIcon iconName="close" onClick={onClose} className="close-btn" />
+
```

Comment:
There are components that we should use for building modals like CWModalHeader, CWModalBody, CWModalFooter. Thanks to them we make sure that the paddings are the same for each modals

Reply:
I used the CWModalBody and CWModalFooter components, but wasn't able to get the proper layout/styling/variant working with CWModalHeader, it is still custom atm.

<img width="599" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/51641047/729a0202-2a1f-4191-a59d-d02af9045b72">


I think there are 2 options
1. make CWModalHeader configurable in many different ways to support the new header layout, font sizes, alignments, x-icon alignment, optional logo, label alignments, and make sure all of these combinations work consistently (this would consume time, and in favor of not breaking anything, i didn't do it right now)
2. use a custom header implementation in 1-off scenarios (like this one)

and since this is a 1-off instance of this header I believe a custom implementation works here. 

Plz lmk your thoughts, do you still think we should follow 1 here? If yes, I would be happy to create a followup and do it after the sign-in milestone.

cc: @masvelio

Reply:
nope, I think #2 is good in this case. We don't want super fancy generic header that is too complicated

Reply:
Awesome, resolving this.